p2p/peerManager/peerManager.go

```
@@ -241,3 +299,46 @@ func (pm *BasicPeerManager) UnprotectPeer(peer p2p.PeerID) {
 func (pm *BasicPeerManager) BanPeer(peer p2p.PeerID) {
 	pm.BlockPeer(peer)
 }
+
+func (pm *BasicPeerManager) Stop() error {
+	var closeErrors []string
+
+	if err := pm.BasicConnMgr.Close(); err != nil {
+		closeErrors = append(closeErrors, err.Error())
+	}
+	if err := pm.bestPeersDB.Close(); err != nil {
+		closeErrors = append(closeErrors, err.Error())
+	}
+	if err := pm.responsivePeersDB.Close(); err != nil {
+		closeErrors = append(closeErrors, err.Error())
+	}
+	if err := pm.allPeersDB.Close(); err != nil {
+		closeErrors = append(closeErrors, err.Error())
+	}
+
+	if len(closeErrors) > 0 {
+		return errors.New("failed to close some resources: " + strings.Join(closeErrors, "; "))
+	}
+
+	return nil
```

We should return the errors here

---

Oh how did I miss